pxtlib/emitter/ir.ts

```diff
@@ -310,6 +312,38 @@ namespace ts.pxt.ir {
                 }
             }
         }
+
+        get isTemporary(): boolean {
+            return false
+        }
+    }
+
+    //Cells that represent variables that are generated by the compiler as temporaries
+    //The user cannot access these cells from JS or blocks
+    export class UnnamedCell extends Cell {
+        private static unnamedCellCounter = 0
+        private uid: number
+
+        constructor(public index: number, public owningProc: Procedure) {
+            super(index, null, null)
+            this.uid = UnnamedCell.unnamedCellCounter++
+        }
+
+        getName() {
+            return "unnamed" + this.uid
+        }
+
+        uniqueName() {
+            return this.getName() + "___" + this.index
```

Comment:
This needs to be something like "___U", otherwise there is some (small) risk it will clash with normal variables